### DISCUSSION TITLE
writing plugins

### DISCUSSION THREAD

**Discussion by riaanmarx** [winder/Universal-G-Code-Sender#1657] [Help] [answered]
Being a software developer myself, I am always thinking about customizing and adding to a product.
Is there any documentation about how to start building plugins for UGS (on Windows OS preferably).

**Comment by riaanmarx** [comment, marked as answer]
found it....
[https://winder.github.io/ugs_website/dev/plugin/](url)

**Reply by dlpgraphics** [reply]
This link gets me 404. Is there an updated link?

**Reply by breiler** [reply]
The link had the right text but wrong url in the actual link.
https://winder.github.io/ugs_website/dev/plugin/

**Reply by breiler** [reply]
To get started with just compiling the software have a look here:
https://github.com/winder/Universal-G-Code-Sender#development
